src/omnicore/rpc.cpp

```diff
@@ -968,87 +964,119 @@ Value gettradessince_MP(const Array& params, bool fHelp)
-    for (md_PricesMap::iterator it = prices.begin(); it != prices.end(); ++it)
-    {
-      md_Set & indexes = (it->second);
-      for (md_Set::iterator it = indexes.begin(); it != indexes.end(); ++it)
-      {
-          CMPMetaDEx obj = *it;
+    // TODO: sort response array on confirmations attribute and crop response to (count) most recent transactions
+    return response;
+}
 
-          bool filter = 1;
+Value gettradehistoryforpair_OMNI(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+            "gettradehistory_MP\n"
+            "\nAllows user to retrieve MetaDEx trade history for the specified market\n"
+            "\nArguments:\n"
+            "1. propertyid           (int, required) the first side of the pair\n"
+            "2. propertyid           (int, required) the second side of the pair\n"
+            "3. count                (int, optional) number of trades to retrieve (default: 10)\n"
+        );
 
-          //this filter, the first part is filtering by two currencies, the second part is filtering by the first only
-          filter = (obj.getAddr() == address) && 
-                   ( ( ( filter_by_one && (obj.getProperty() == propertyIdFilter) == 1 ) ) || !filter_by_one ) && 
-                   (vMetaDexObjects.size() < number_trades);
-          if ( filter ) {
-            vMetaDexObjects.push_back(obj);
-          }
-      }
-    }
-  }
+    Array response;
 
-  }
+    // obtain property identifiers for pair & check valid parameters
+    uint32_t propertyIdSideA = 0, propertyIdSideB = 0;
+    int64_t tmpPropertyIdSideA = params[0].get_int64();
+    int64_t tmpPropertyIdSideB = params[1].get_int64();
+    if (tmpPropertyIdSideA < 1 || tmpPropertyIdSideA > 4294967295 || tmpPropertyIdSideB < 1 || tmpPropertyIdSideB > 4294967295)
+         throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid property identifier");
+    propertyIdSideA = boost::lexical_cast<uint32_t>(tmpPropertyIdSideA);
+    propertyIdSideB = boost::lexical_cast<uint32_t>(tmpPropertyIdSideB);
+    if (!_my_sps->hasSP(propertyIdSideA) || !_my_sps->hasSP(propertyIdSideB))
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Property identifier does not exist");
+    if ((propertyIdSideA != 1 && propertyIdSideA != 2 && propertyIdSideB != 1 && propertyIdSideB != 2) || propertyIdSideA == propertyIdSideB)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid trade pair");
+
+    // obtain count parameter if supplied
+    int64_t count = 10;
+    if (params.size() > 2) count = params[2].get_int64();
+    if (count < 1) // TODO: do we need a maximum here?
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid value for count parameter");
+
+    // metadex maps are irrelevant for this call, we are looking for completed trades which are stored in tradelistdb
+    t_tradelistdb->getTradesForPair(propertyIdSideA, propertyIdSideB, response);
```

Comment:
```c++
{
    LOCK(cs_tally):
    t_tradelistdb->getTradesForPair(propertyIdSideA, propertyIdSideB, response);
}
```

Other than that, the `threadSafe` flag for this call can also be removed in `rpcserver.cpp`.